### PR TITLE
Add IT for dolphinscheduler upgrade

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

Make sure the upgrade tools can execute successfully, after this PR merged we can add more assert case.

## Brief change log

Add two test case to execute the upgrade tools and initialize tools.

## Verify this pull request

Add IT case.

### CODE REVIEW

**Comment by zhongjiajie** [comment]
Some UT still failed

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/mysql/dolphinscheduler_ddl.sql [hunk 1]
```diff
@@ -53,8 +53,8 @@ BEGIN
 ALTER TABLE t_ds_process_definition ADD COLUMN `execution_type` tinyint(4) DEFAULT '0' COMMENT 'execution_type 0:parallel,1:serial wait,2:serial discard,3:serial priority';
 END IF;
 END;
- d//
- delimiter ;
+d//
+delimiter ;
```
Comment (zhongjiajie) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/mysql/dolphinscheduler_ddl.sql]: are those changes for bug fixes or just formatted?

**Review by zhongjiajie — COMMENTED** [review]
Others LGTM only if ci pass

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/mysql/dolphinscheduler_ddl.sql (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/mysql/dolphinscheduler_ddl.sql]: This is a bug for current parser, I fix this in parser and rollback this file.

**Comment by ruanwenjun** [comment]
> Some UT still failed

It seems caused by current byte-buddy doesn't support jdk17, I upgrade the byte-buddy version to fix this.

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/mysql/dolphinscheduler_ddl.sql [hunk 2]
```diff
@@ -445,9 +445,7 @@ BEGIN
            AND TABLE_SCHEMA=(SELECT DATABASE())
            AND COLUMN_NAME ='task_name')
    THEN
-       ALTER TABLE `t_ds_fav_task` MODIFY COLUMN `id` bigint NOT NULL AUTO_INCREMENT;
        ALTER TABLE `t_ds_fav_task` DROP COLUMN `task_name`;
-       ALTER TABLE `t_ds_fav_task` ADD COLUMN `task_type` varchar(64) NOT NULL COMMENT 'favorite task type name';
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/mysql/dolphinscheduler_ddl.sql:450]: `id`, `task_type` already exist in `t_ds_fav_task`

Comment (zhongjiajie) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.0_schema/mysql/dolphinscheduler_ddl.sql:450]: CI failed due to schema DDL not sync when we try to upgrade from 3.1.8

```sql
ALTER TABLE `t_ds_fav_task` ADD COLUMN `task_type` varchar(64) NOT NULL COMMENT "favorite task type name";
```

**Comment by zhongjiajie** [comment]
FYI, CI failed due to schema DDL not sync when we try to upgrade from 3.1.8

```sql
ALTER TABLE `t_ds_fav_task` ADD COLUMN `task_type` varchar(64) NOT NULL COMMENT "favorite task type name";
```

**Comment by zhongjiajie** [comment]
and I am not sure why K8S test failed. Others looking good 👍

**Comment by ruanwenjun** [comment]
> and I am not sure why K8S test failed. Others looking good 👍

K8S test has be fixed at #15053

**Review by zhongjiajie — APPROVED** [review]

**Pull request merged**